block-layer/codecs/dag-cbor.md

```diff
@@ -37,93 +39,103 @@ In DAG-CBOR, map keys must be strings, as defined by the [IPLD Data Model]. Othe
+ * Any `Number` serialized by the JavaScript CBOR encoder relies on a whole-number check (i.e. `Number.isInteger()`, roughly `x % 1 === 0`) to determine whether it should be encoded as an integer or a float.
+ * Any float deserialized by the JavaScript CBOR decoder that does not have a fractional component will be indistinguishable from an integer to a JavaScript program and may not round-trip to the same bytes if originally produced by non-JavaScript code.
+ * Any `Number` greater than `Number.MAX_SAFE_INTEGER` or less than `Number.MIN_SAFE_INTEGER` cannot be properly inspected for its whole-number status and is therefore encoded by the JavaScript CBOR encoder as float regardless of whether it is a whole-number or has a fractional component. [`BigInt`] should be used for [@ipld/dag-cbor] when dealing with integers outside of the safe range to ensure proper handling.
 
 [IPLD Data Model]: ../../data-model-layer/data-model.md
-[Concise Binary Object Representation (CBOR)]: https://tools.ietf.org/html/rfc7049
+[Concise Binary Object Representation (CBOR)]: https://cbor.io/
+[RFC 8949]: https://tools.ietf.org/html/rfc8949
+[RFC 7049]: https://tools.ietf.org/html/rfc7049
 [IPLD Data Model Kinds]: ../../data-model-layer/data-model.md#kinds
 [Links]: ../../data-model-layer/data-model.md#link-kind
-[CIDs]: ../CID.md
+[CID]: ../CID.md
 [Multibase]: https://github.com/multiformats/multibase
-[section 2.4 of the CBOR specification]: https://tools.ietf.org/html/rfc7049#section-2.4
-[section 3.9 of the CBOR specification]: https://tools.ietf.org/html/rfc7049#section-3.9
-[section 3.1 of the CBOR specification]: https://tools.ietf.org/html/rfc7049#section-3.1
-[borc]: https://github.com/dignifiedquire/borc
-[dag-cbor]: https://github.com/ipld/js-ipld-dag-cbor/
-[refmt]: https://github.com/polydawn/refmt/
-[ipld-cbor]: https://github.com/ipfs/go-ipld-cbor
-[ipld-prime]: http://github.com/ipld/go-ipld-prime
-[ipld]: https://github.com/ipld/js-ipld
-[@ipld/block]: https://github.com/ipld/js-block
+[CBOR Tags Registry]: https://www.iana.org/assignments/cbor-tags/cbor-tags.xhtml
+[IPLD content identifiers (CIDs) in CBOR]: https://github.com/ipld/cid-cbor/
+[section 3.4 of RFC 8949]: https://tools.ietf.org/html/rfc8949#section-3.4
+[section 4.2 of RFC 8949]: https://tools.ietf.org/html/rfc8949#section-4.2
+[section 3.9 of RFC 7049]: https://tools.ietf.org/html/rfc7049#section-3.9
+[Simple Values]: https://tools.ietf.org/html/rfc8949#section-2.1
+[section 5.1 of RFC 8949]: https://tools.ietf.org/html/rfc8949#section-5.1
+[@ipld/dag-cbor]: https://github.com/ipld/js-dag-cbor/
+[multiformats]: https://github.com/multiformats/js-multiformats/
+[`BigInt`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
```

Comment:
TIL these can contain special markdown characters

Reply:
I'm only using foot-links here because they're already in this doc, otherwise I _hate_ this style of MD linking

Reply:
Really? I actually find it quite tidy and nice for reading the "plaintext English" without having to render the doc.